### PR TITLE
feat: add ci/cd lambda for deployment

### PR DESCRIPTION
**Pull request by dominikeinkemmer**
Add CI/CD via github actions to deploy to lambda on push to master.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: deploy to lambda
+on:
+  push:
+    branches:
+      - add-ci-cd
```
Comment (dominikeinkemmer) [inline .github/workflows/deploy.yml]: temporary for testing

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - uses: actions/checkout@v1
+      - name: Install zip
+        run: sudo apt install zip
+      - name: create zip bundle
+        run: zip -qq -r ./bundle.zip ./api-dynamodb-ingest
+      - name: default deploy
+        uses: appleboy/lambda-action@master
+        with:
+          aws_access_key_id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```
Comment (rashnil-git) [inline .github/workflows/deploy.yml:23]: are you storing keys?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+          aws_region: us-east-2
+          function_name: findthecluster-surveyapp-dynamo-ingest
+          zip_file: bundle.zip
```
Comment (rashnil-git) [inline .github/workflows/deploy.yml:26]: We should not change the folder structure for the lambda file. There will be multiple lambda functions. So, its good to have a single lambda folder and under that multiple files for each function. This above flow from actions is good, but we can modify a little to make it flexible.

**Review by rashnil-git — CHANGES_REQUESTED** [review]
Thanks for doing this. good work. should be good after minor changes and some testing. Recommend testing with a dummy lambda before changing existing function.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (dominikeinkemmer) [reply .github/workflows/deploy.yml:23]: on the secrets of the github repository yes

Comment (dominikeinkemmer) [reply .github/workflows/deploy.yml:23]: the key is for my user right now so if that is deactivated probably this will become invalid

**Comment by dominikeinkemmer** [comment]
@rashnil-git changed it a bit, now it only zips the specific file of the lambda function and deploys it.

I tested it with this function:
https://us-east-2.console.aws.amazon.com/lambda/home?region=us-east-2#/functions/test-function-auto-deploy?tab=configuration

and it works :)

**Comment by rashnil-git** [comment]
Great thanks:)

**Review by rashnil-git — APPROVED** [review]

**Pull request merged**